worlds/sc2/rules.py

```diff
@@ -1464,13 +1707,15 @@ def night_terrors_requirement(self, state: CollectionState) -> bool:
                     )
                 )
             )
+            and self.terran_army_weapon_armor_upgrade_count(state) >= 2
```

Comment:
Again, this function will check for all upgrades even if the intended comp is firebat/liberator, or hellbat/vulture, requiring fewer upgrade types. I'd also say that siege tanks and liberators would have lower upgrade requirements because they likely oneshot infesteds with or without them. Vultures would largely be using spider mine damage, so wouldn't benefit from upgrades.

Finally, just noticed that terran bio healing is required even if firebats/hellions are not in the comp, replaced by perdition | PF on advanced tactics.

I almost wonder if it's simpler to just pull liberators out as a viable infested-stomper on their own, and getting rid fo the firebat/hellion/HERC/vulture logic. In vanilla, this mission is often dealt with using mass goliath or mass lib or mass tank. Another way to reformat it to keep something like these comps around is to break things up into multiple statements:

```
backline = liberator or marine(+2) or goliath (+1) or banshee or warhound or widow mine or marauder+concussive or spectre+adv.tactics
frontline = (
    (
      firebat (+1armour) or herc (+1armour) or marauder(+1armour)+kinetic foam or hellbat (+1armour)
    ) and terran_bio_heal
    or perdition or vulture or raven+mines+adv.tactics
)
return common unit and competent_aa and siege tank or (viking+shredder+(+2air)) or backline+frontline
```